# Chi-square test

In [122]:
# H0=Customer's satisfied with smarthome devices purchased (null hypotheses, +ve assumption)
# Ha=Customer's not satisfied with smarthome devices purchased(alternative hypotheses)

In [123]:
from scipy.stats.contingency import chi2_contingency

In [124]:
data=[[50,70],[80,100],[60,90],[30,50],[20,50]]

In [125]:
from scipy.stats.contingency import chi2_contingency
chi,p,df,expected=chi2_contingency(data)

In [126]:
print(chi,p,df,expected)

5.638227513227513 0.22784371130697179 4 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [127]:
alpha=0.05
print("p value is" +str(p))
if p<alpha:
    print(" Customer satisfied with smarthome device purchased(Rejecct H0)")
else:
    print("Customer not satisfied with smarthome device purchased(Fail to reject H0)")

p value is0.22784371130697179
Customer not satisfied with smarthome device purchased(Fail to reject H0)


In [119]:
#critical value=9.488  ,from chi_sq table

In [120]:
import scipy.stats as stats

In [121]:
stats.chi2.ppf(1-0.05,4) #stats.chi2.ppf(1-alpha,df)

9.487729036781154

### p>alpha=fail to reject H0
### 0.2278>o.o5=fail to reject H0,   i.e Customers satisfied with smarthome devices purchased

### critical>statistical value=fail to reject H0
### 9.4877>5.6382= fail to reject H0,  i.e Customers satisfied with smarthome devices purchased

### conclusion:- Customers satisfied with smarthome devices(Smart Thermostats,Smart Lights) purchased

# Hypothesis testing

#Bombay hospitality Ltd. operates a franchise model for producing exotic Norwegian dinners throughout New England. The operating cost for a franchise in a week (W) is given by the equation W = $1,000 + $5X, where X represents the number of units produced in a week. Recent feedback from restaurant owners suggests that this cost model may no longer be accurate, as their observed weekly operating costs are higher.

# Null Hypothesis (H0): The theoretical cost model is accurate.
# Alternative Hypothesis (Ha): The theoretical cost model is inaccurate.



In [29]:
import pandas as pd
import numpy as np
import math
import scipy.stats as stats

In [36]:
n=25 # Sample size
sample_mean= 3050 # Sample mean weekly cost (Rs.)
theoretical_mean = 4000 # Theoretical mean cost (1000$ + 5 * units produced)(1000+5*600)
std_dev=125 # Standard deviation(5*25)
alpha=0.05 # Significance level

In [31]:
# calculating t staticcal value

In [32]:
t_statistics= (3050-4000)/(125/np.sqrt(25))
t_statistics


-38.0

In [33]:
# Calculating the t critical value

In [34]:
t_critical=stats.norm.ppf(1-0.05/2) #2_tail 
t_critical

1.959963984540054

In [37]:
# Print results
print("Sample Mean Weekly Cost:", sample_mean, "Rs.")
print("Theoretical Mean Weekly Cost:", theoretical_mean, "Rs.")
print("Test Statistic (t):", t_statistics)
print("Critical Value (alpha =", alpha, "):", t_critical)


Sample Mean Weekly Cost: 3050 Rs.
Theoretical Mean Weekly Cost: 4000 Rs.
Test Statistic (t): -38.0
Critical Value (alpha = 0.05 ): 1.959963984540054


In [38]:
print("t_ Statistic (t):", t_statistics)
print("Critical Value (+/-):", t_critical)

t_ Statistic (t): -38.0
Critical Value (+/-): 1.959963984540054


In [39]:
if abs(t_statistics) > t_critical:
    
  print("Reject Null Hypothesis. The observed cost deviates significantly from the model.")

else:
  print("Fail to Reject Null Hypothesis. There's insufficient evidence to reject the model.")

Reject Null Hypothesis. The observed cost deviates significantly from the model.
